### PR TITLE
[CHORE] Fix dangerjs function for version check

### PR DESCRIPTION
**Pull request by miodrage**
**JIRA**
https://capmo-team.atlassian.net/browse/IES-1000

**Summary**

https://we-are-capmo.slack.com/archives/D9ZVCCM7E/p1621284946007700

FYI: In the @capmo/config lib I had an issue with the DangerJs’ function which checks for a package version update. See the picture.
In order to check what is going on, I opened our capmo config library and saw this comment above the function in charge of handling the version check.
TODO: Improve function to look for version bump vs checking files changed
So I went forward and fixed a few things along the way, instead of force merging my PR.
Things which I added in this PR are:
CircleCi added to the repo which only runs DangerJS
GitHub pull request template has been added
DangerJS added to the repo (as already mentioned above)
A better way to check for a package version update

### CODE REVIEW

**Comment by capmo-jarvis** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  `node` failed.
  </td>
    </tr>
  </tbody>
</table>



### Log


```sh
Error:  Error: Could not find a 'dangerfile.js' or 'dangerfile.ts' in the current working directory (/home/circleci/app).
    at Object.dangerfilePath (/home/circleci/app/node_modules/danger/distribution/commands/utils/fileUtils.js:28:11)
    at /home/circleci/app/node_modules/danger/distribution/commands/danger-runner.js:92:42
    at step (/home/circleci/app/node_modules/danger/distribution/commands/danger-runner.js:34:23)
    at Object.next (/home/circleci/app/node_modules/danger/distribution/commands/danger-runner.js:15:53)
    at fulfilled (/home/circleci/app/node_modules/danger/distribution/commands/danger-runner.js:6:58)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)

```

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against f98d25e9b6f166dc0812d4467661723af3291a6e
</p>

**Comment by capmo-jarvis** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Please ensure the package version has been bumped in accordance with [Semantic Versioning](https://semver.org/)
and run `npm i` to update `package-lock.json`

  </td>
    </tr>
  </tbody>
</table>




<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against bd55a86f1b052f6005bd559ea7b6fe47f23ebd36
</p>

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@capmo/config",
-  "version": "1.11.7",
+  "version": "1.11.8",
```
Comment (miodrage) [inline package.json:3]: Please see the package-lock.json file change.

**Review by vahid-capmo — APPROVED** [review]

File: dangerjs/version.js [hunk 2]
```diff
@@ -1,26 +1,14 @@
...
-  const lockfileChanged = modifiedFiles.includes('package-lock.json')
-
-  if (packageChanged && !lockfileChanged) {
-    return packageChangedNotLockError.join('')
-  }
+function checkForVersionUpdate ({ versionValidation, packageJsonDiff }) {
+  if (!versionValidation) return
 
-  if (packageChanged) return
+  if(packageJsonDiff && packageJsonDiff.added.includes(PACKAGE_JSON_VERSION_PROP_QUERY) && packageJsonDiff.removed.includes(PACKAGE_JSON_VERSION_PROP_QUERY)) return
```
Comment (GregoryEDev) [inline dangerjs/version.js:11]: Please extract these to make it a little more readable

File: .circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,55 @@
+version: 2.1
```
Comment (GregoryEDev) [inline .circleci/config.yml:1]: Why are we adding the Circle CI stuff here?

Comment (miodrage) [reply .circleci/config.yml:1]: To apply dangerjs rules to this repo's PR as well.

File: dangerjs/version.js [hunk 4]
```diff
@@ -1,26 +1,27 @@
...
+  const packageJson = {
+    isNewVersionSet: packageJsonDiff && packageJsonDiff.added.includes(PACKAGE_JSON_VERSION_PROP_QUERY),
+    isOldVersionRemoved: packageJsonDiff && packageJsonDiff.removed.includes(PACKAGE_JSON_VERSION_PROP_QUERY)
+  }
 
-  if (packageChanged && !lockfileChanged) {
-    return packageChangedNotLockError.join('')
+  const packageLockJson = {
+    isNewVersionSet: packageLockJsonDiff && packageLockJsonDiff.added.includes(PACKAGE_JSON_VERSION_PROP_QUERY),
+    isOldVersionRemoved: packageLockJsonDiff && packageLockJsonDiff.removed.includes(PACKAGE_JSON_VERSION_PROP_QUERY)
```
Comment (GregoryEDev) [inline dangerjs/version.js:18]: Since the old version removal is automatic in the Diff, do you think it really necessary to check for the removal?

**Comment by capmo-jarvis** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Danger failed to run `dangerfile.js`.
  </td>
    </tr>
  </tbody>
</table>



## Error SyntaxError
```
Invalid regular expression: /"version": (?'MAJOR'(?:0|(?:[1-9]\d*)))\.(?'MINOR'(?:0|(?:[1-9]\d*)))\.(?'PATCH'(?:0|(?:[1-9]\d*)))(?:-(?'prerelease'[0-9A-Za-z-]+(\.[0-9A-Za-z-]+)*))?(?:\+(?'build'[0-9A-Za-z-]+(\.[0-9A-Za-z-]+)*))?$/: Invalid group
SyntaxError: Invalid regular expression: /"version": (?'MAJOR'(?:0|(?:[1-9]\d*)))\.(?'MINOR'(?:0|(?:[1-9]\d*)))\.(?'PATCH'(?:0|(?:[1-9]\d*)))(?:-(?'prerelease'[0-9A-Za-z-]+(\.[0-9A-Za-z-]+)*))?(?:\+(?'build'[0-9A-Za-z-]+(\.[0-9A-Za-z-]+)*))?$/: Invalid group
    at Object.<anonymous> (/home/circleci/app/dangerjs/version.js:6:21)
    at Module._compile (internal/modules/cjs/loader.js:999:30)
    at Object.customModuleHandler (/home/circleci/app/node_modules/danger/distribution/runner/runners/inline.js:129:28)
    at Module.load (internal/modules/cjs/loader.js:863:32)
    at Function.Module._load (internal/modules/cjs/loader.js:708:14)
    at Function._module2.default._load (/home/circleci/app/node_modules/override-require/dist/overrideRequire.js:43:25)
    at Module.require (internal/modules/cjs/loader.js:887:19)
    at require (internal/modules/cjs/helpers.js:74:18)
    at Object.<anonymous> (/home/circleci/app/dangerjs/index.js:8:35)
    at Module._compile (internal/modules/cjs/loader.js:999:30)
```
### Dangerfile
```
1| /**
2|  * DangerJS does not allow for the packages to be // Removed imported inside the Capmo Config
3| //  * Unfortunately this is comp// Removed require; Original:  letely ignored at runtime and needs to be imported in
4| // //  * the dangerfile. To overcome this we import Danger in the local dangerfile of each
5| // //  * Repository and disable no-unused-vars for the single line.
----------------------^
6| // //  */
7| // // // eslint-disable-next-line @typescript-eslint/no-unused-vars
8| // // const { fail, danger } = require('danger');
9| const { dangerJs } = require('./dangerjs');
```
  
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against c639c584583c337a91cbf83382a932aa21ee338c
</p>

**Comment by capmo-jarvis** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  `node` failed.
  </td>
    </tr>
  </tbody>
</table>



### Log


```sh
Error:  ReferenceError: PACKAGE_JSON_VERSION_PROP_QUERY is not defined
    at checkForVersionUpdate (/home/circleci/app/dangerjs/version.js:10:77)
    at dangerJs (/home/circleci/app/dangerjs/index.js:28:24)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
danger-results://tmp/danger-results.json
```

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against c79e483c84955bb500518dcc3a6123dc7158a292
</p>

**Comment by capmo-jarvis** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Please ensure the package version has been bumped in accordance with [Semantic Versioning](https://semver.org/)
  </td>
    </tr>
  </tbody>
</table>




<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against b477be5cb7a031da41dc5d5754c48a0ddeb829ce
</p>

File: dangerjs/version.js [hunk 5]
```diff
@@ -1,28 +1,30 @@
...
-    return packageChangedNotLockError.join('')
-  }
+  const packageJsonNewVersion = packageJsonDiff.added.match(REGEX_VERSION_QUERY)[0];
+  const packageLockJsonNewVersion = packageLockJsonDiff.added.match(REGEX_VERSION_QUERY)[0];
 
-  if (packageChanged) return
+  if(packageJsonNewVersion !== packageLockJsonNewVersion) return errors.versionMissmatch;
 
-  return versionNotBumpedError.join('')
+  return;
```
Comment (GregoryEDev) [inline dangerjs/version.js:27]: Is this necessary since the return is implied?

File: dangerjs/version.js [hunk 6]
```diff
@@ -1,28 +1,28 @@
...
+
+function checkForVersionUpdate ({ versionValidation, packageJsonDiff, packageLockJsonDiff }) {
   if (!versionValidation) return
 
-  const packageChanged = modifiedFiles.includes('package.json')
-  const lockfileChanged = modifiedFiles.includes('package-lock.json')
+  const isPackageJsonChanged = packageJsonDiff && packageJsonDiff.added;
+  const isPackageLockJsonChanged = packageLockJsonDiff && packageLockJsonDiff.added;
+
+  const isNewVersionSet = isPackageJsonChanged && packageJsonDiff.added.match(REGEX_VERSION_QUERY);
```
Comment (GregoryEDev) [inline dangerjs/version.js:16]: Could you please remove all semi-colons as per the Prettier config

Comment (miodrage) [reply dangerjs/version.js:16]: 😢

**Review by GregoryEDev — APPROVED** [review]

**Pull request merged**